fable-test-app/src/pages/ViewHolidays.tsx

```diff
@@ -1,16 +1,135 @@
 
 // Components (internal)
-import { DateModified, Heading } from '../components';
+import { DateModified, Details, Heading, Select, StyledLink, Text } from '../components';
+import { holidayData } from '../data/holidayData';
+import { formatDate } from '../utils/utils';
+
+const ViewHolidays = () => {
+  const { provinceId } = useParams<{ provinceId: string }>();
+
+  // State variables
+  const [holidays, setHolidays] = useState([]);
+  const [year, setYear] = useState<string>('2024');
+  const [yearsList] = useState<string[]>(['2022', '2023', '2024', '2025', '2026']);
+  const [nextHoliday, setNextHoliday] = useState(null);
+
+  // Find data for current province/territory
+  const province = holidayData.find(p => p.id === provinceId);
+
+  useEffect(() => {
+    if (province) {
+      const endpointForYear = `${province.endpoint}&year=${year}`;
+
+      axios.get(endpointForYear)
+        .then(({ data }) => {
+          const holidaysData = provinceId === 'federal' ? data.holidays : data.province.holidays;
+          const nextHolidayData = provinceId === 'federal' ? data.nextHoliday : data.province.nextHoliday;
+
+          setHolidays(holidaysData);
+          setNextHoliday(nextHolidayData || null);
+        })
+        .catch(error => {
+          console.error("There was an error fetching the holidays!", error);
+        });
+    }
+  }, [province, year, provinceId]);
+
+  if (!province) {
+    return <p>Province or territory not found.</p>;
+  }
+
+  // Calculate days until next holiday
+  const calcNextHoliday = (dateString) => {
+    const today = new Date();
+    const holidayDate = new Date(dateString);
+
+    return Math.floor((holidayDate.getTime() - today.getTime()) / (1000 * 3600 * 24));
+  };
+
+  const daysToNextHoliday = nextHoliday ? calcNextHoliday(nextHoliday.date) : null;
 
-const ViewHolidays: React.FC = () => {
   return (
     <section>
-      <Heading tag="h1">Nationwide holidays</Heading>
+      <Heading tag="h1">{province.name} holidays</Heading>
+
+      <Select
+        selectId="select-province-year"
+        label="Calendar year"
+        hint="Select the year of holidays you want to view."
+        name="province-year"
+        value={year}
+        onInput={(e) => setYear(e.target.value)}
+      >
+        {yearsList.map(yearOption => (
+          <option key={yearOption} value={yearOption}>
+            {yearOption}
+          </option>
+        ))}
+      </Select>
+
+      {nextHoliday && (
+        <div className="d-flex bg-primary md:align-items-center align-items-start text-light mb-450 md:px-450 px-250 py-200">
+          <img
+            className="d-inline-block me-400"
+            src="/icons/icon-calendar.svg"
+            alt="Calendar icon with a clock in the bottom right corner."
+          />
+          <Text textRole="light" marginBottom="0">
+            <strong>Next holiday is {nextHoliday.nameEn} — that's {daysToNextHoliday} days away</strong>
+          </Text>
+        </div>
+      )}
```

Comment:
This would be nice to have in a separate component - where the computation to get the next holiday is stored there separate from this page